irods_specific_implementation/irods_rest_access_implementation.h

```diff
@@ -1,147 +1,168 @@
-
 #include "irods_rest_api_base.h"
 #include "filesystem.hpp"
+#include "pistache/http_headers.h"
+#include "pistache/optional.h"
 #include "rodsClient.h"
 #include "irods_random.hpp"
 
+#include "fmt/format.h"
+
 #include <chrono>
-#include <ctime>
+#include <string_view>
 
 // this is contractually tied directly to the swagger api definition, and the below implementation
 #define MACRO_IRODS_ACCESS_API_IMPLEMENTATION \
     Pistache::Http::Code code; \
     std::string message; \
-    std::tie(code, message) = irods_access_(headers.getRaw("Authorization").value(), path.get(), base); \
+    std::tie(code, message) = irods_access_(headers, path.get(), base, use_count, expires_after); \
     response.send(code, message);
 
-namespace irods::rest {
-    namespace sc = std::chrono;
-    using sclk = sc::system_clock;
-
-    namespace fs   = irods::experimental::filesystem;
-    namespace fcli = irods::experimental::filesystem::client;
-    using     fsp  = fs::path;
-
+namespace irods::rest
+{
     // this is contractually tied directly to the api implementation
     const std::string service_name{"irods_rest_cpp_access_server"};
 
-    class access : public api_base {
-        public:
+    class access : public api_base
+    {
+    public:
+        access()
+            : api_base{service_name}
+        {
+            logger_->trace("Endpoint [{}] initialized.", service_name);
+        }
 
-            access() : api_base{service_name}
-            {
-                // ctor
-            }
+        std::tuple<Pistache::Http::Code &&, std::string>
+        operator()(const Pistache::Http::Header::Collection& _headers,
+                   const std::string& _logical_path,
+                   const std::string& _base_url,
+                   const Pistache::Optional<std::string>& _use_count,
+                   const Pistache::Optional<std::string>& _expires_after)
```

Comment:
consider including the word `seconds` in the name. for example:
- _expires_in_seconds
- _expires_after_n_seconds
- _seconds_before_expiry
- **_seconds_until_expiration**
- _good_for_number_of_seconds
- _seconds_until_bad

Reply:
use this for the URL parameter name too.

Reply:
done